addons/project_timesheet_holidays/models/account_analytic.py

```diff
@@ -10,8 +10,27 @@ class AccountAnalyticLine(models.Model):
 
     holiday_id = fields.Many2one("hr.leave", string='Leave Request')
     global_leave_id = fields.Many2one("resource.calendar.leaves", string="Global Time Off", ondelete='cascade')
+    task_id = fields.Many2one(domain="[('company_id', '=', company_id), ('project_id.allow_timesheets', '=', True),"
+        "('project_id', '=?', project_id), ('is_timeoff_task', '=', False)]")
 
     @api.ondelete(at_uninstall=False)
     def _unlink_except_linked_leave(self):
         if any(line.holiday_id for line in self):
-            raise UserError(_('You cannot delete timesheets linked to time off. Please, cancel the time off instead.'))
+            raise UserError(_('You cannot delete timesheets that are linked to time off requests. Please cancel your time off request from the Time Off application instead.'))
+
+    @api.model_create_multi
+    def create(self, vals_list):
+        if not self.env.su:
+            tasks = self.env['project.task'].browse([
+                vals['task_id']
+                for vals in vals_list
+                if vals.get('task_id')
+            ])
+            if any(tasks.mapped('is_timeoff_task')):
```

Comment:
We could just make a search (with limit=2) and check the number of returned ids in order to prevent fetching all the record data right?

Reply:
If user create timesheets in which it contains more than 3 task and third task is time off task then this suggestion may skip that use case. I don't khow how to produce this scenario. Thanks :slightly_smiling_face: